### PR TITLE
Add signal to support additional foreign keys on Sample

### PR DESCRIPTION
**Pull request by awenocur**
This allows the delete-sample command to be extended, to support models with foreign keys pointing to Sample.

### CODE REVIEW

File: vdw/signals.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from django.dispatch import Signal
+
+pre_delete_sample = Signal(providing_args=["pk_set", "cursor"])
```
Comment (bruth) [inline vdw/signals.py]: Change strings to use single quotes to be consistent with the rest of the codebase.

File: vdw/samples/management/subcommands/delete_project.py [hunk 2]
```diff
@@ -111,6 +113,9 @@ def handle(self, *args, **options):
                 ''', [project_id])
 
                 # Remove samples
+                pre_delete_sample.send(sender=Project,
```
Comment (naegelyd) [inline vdw/samples/management/subcommands/delete_project.py]: If this is going to use the `pre_delete_sample` signal then the `pk_set` argument should be the sample IDs in the project being deleted. If you are going to send the project ID then change this to use a new `pre_delete_project` signal.

**Comment by naegelyd** [comment]
Please squash this into a single commit

File: vdw/samples/management/subcommands/delete_project.py [hunk 3]
```diff
@@ -111,6 +113,9 @@ def handle(self, *args, **options):
                 ''', [project_id])
 
                 # Remove samples
```
Comment (naegelyd) [inline vdw/samples/management/subcommands/delete_project.py:119]: Reorder these lines:

``` python
pre_delete_project.send(
    sender=Project, pk=project_id, cursor=cursor)

# Remove samples
cursor.execute('''
    DELETE FROM sample WHERE project_id = %s
''', [project_id])
```

File: vdw/samples/management/subcommands/delete_sample.py [hunk 4]
```diff
@@ -70,6 +71,9 @@ def handle(self, *args, **options):
                 '''.format(columns), valid_ids)
 
                 # Remove sample
+                pre_delete_sample.send(sender=Sample,
```
Comment (naegelyd) [inline vdw/samples/management/subcommands/delete_sample.py]: Nit, move all arguments to the next line to become:

``` python
pre_delete_sample.send(
    sender=Sample, pk_set=valid_ids, cursor=cursor)
```

File: vdw/samples/management/subcommands/delete_sample.py [hunk 5]
```diff
@@ -70,6 +71,8 @@ def handle(self, *args, **options):
                 '''.format(columns), valid_ids)
 
                 # Remove sample
```
Comment (naegelyd) [inline vdw/samples/management/subcommands/delete_sample.py]: Remove comment

File: vdw/samples/management/subcommands/delete_sample.py [hunk 6]
```diff
@@ -70,6 +71,8 @@ def handle(self, *args, **options):
...
+                pre_delete_sample.send(
+                    sender=Sample, pk_set=valid_ids, cursor=cursor)
```
Comment (naegelyd) [inline vdw/samples/management/subcommands/delete_sample.py:74]: Add blank line below

**Pull request merged**